pkg/tagger/collectors/ecs_fargate_main.go

```diff
@@ -25,11 +26,12 @@ const (
 
 // ECSFargateCollector polls the ecs metadata api.
 type ECSFargateCollector struct {
-	infoOut      chan<- []*TagInfo
-	expire       *taggerutil.Expire
-	lastExpire   time.Time
-	expireFreq   time.Duration
-	labelsAsTags map[string]string
+	infoOut         chan<- []*TagInfo
+	expire          *taggerutil.Expire
+	lastExpire      time.Time
+	expireFreq      time.Duration
+	labelsAsTags    map[string]string
+	doOnceOrchScope sync.Once
```

Comment:
nit, why don't we embed the struct?
```suggestion
	sync.Once
```

Reply:
I feel pretty strongly that we should encourage using the same `sync.Once` for anything else other than this `orchestrator scope`, if that changes we may want to revisit, But also consider that If we embed it, `Do` becomes an exported method for `ECSFargateCollector` while it's actually more of an implementation detail that we use `sync.Once`.

Reply:
Probably worth a comment for the scope of this once.